components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -416,29 +419,37 @@ func (o *odometry) trackPosition() {
 			centerAngle := (rightDist - leftDist) / o.baseWidth
 
 			// Update the position and orientation values accordingly.
-			o.mu.Lock()
```

Comment:
cc @martha-johnston @JohnN193 I remember we had to put the lock in an odd place, and we were going to look at changing this function's style a little, do we remember why it was so early in this code?

Reply:
@biotinker I think the lock needs to stay here because in line 423 you're accessing `o.orientation` which can be changed in reconfigure

Reply:
and I think it's also needed for lines 434 and 435 because you're accessing `o.position`

Reply:
I agree with that. I think this is actually better restored to its former state; otherwise there could be a race if there is a reconfigure during the calculations.